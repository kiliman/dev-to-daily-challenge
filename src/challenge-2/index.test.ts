import { diamond } from '.'

describe('#2 diamond', () => {
  it('should return null if width is even or negative', () => {
    expect(diamond(2)).toBe(null)
    expect(diamond(-1)).toBe(null)
  })

  it('should return a string as diamond shape with width', () => {
    expect(diamond(3)).toMatchSnapshot()
    expect(diamond(5)).toMatchSnapshot()
    expect(diamond(9)).toMatchSnapshot()
  })
})
