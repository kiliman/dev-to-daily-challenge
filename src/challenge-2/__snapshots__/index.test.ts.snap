// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#2 diamond should return a string as diamond shape with width 1`] = `
" * 
***
 * 
"
`;

exports[`#2 diamond should return a string as diamond shape with width 2`] = `
"  *  
 *** 
*****
 *** 
  *  
"
`;

exports[`#2 diamond should return a string as diamond shape with width 3`] = `
"     *     
    ***    
   *****   
  *******  
 ********* 
***********
 ********* 
  *******  
   *****   
    ***    
     *     
"
`;
